File: 2022/UICPC DIV1/Contests/UICPC Round #13 (Div 1)/solutions/H/zo.py
"""
Solves Zebras and Ocelots
Rob Hochberg
"""

from sys import stdin
S = int(stdin.readline())

numBells = 0
for i in range(S):
    animal = stdin.readline().strip()
    numBells = (numBells << 1) | (1 if animal == 'O' else 0)

print numBells

File: 2022/UICPC DIV1/Contests/UICPC Round #13 (Div 1)/solutions/H/ZO.java
/**
 * Solves Zebras and Ocelots
 * Rob Hochberg
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class ZO{
    public static void main(String args[]) throws Exception {
	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

	// Read the input
	int numCritters = Integer.valueOf(br.readLine());

	char[] critters = new char[numCritters * 3];

	long numBells = 0;
	br.read(critters, 0, 2*numCritters);
	for(int i = 0; i < numCritters; i++){
	    numBells <<= 1;
	    if(critters[2*i] == 'O')
		numBells++;
	}
	System.out.println(numBells);
    }
}
